File: backend/app/schemas/workspaces.py
"""Pydantic schemas for workspace resources."""

from datetime import datetime
from typing import Optional
from uuid import UUID

from pydantic import BaseModel


class WorkspaceCreate(BaseModel):
    name: str
    slug: str
    description: Optional[str] = None
    icon: Optional[str] = None
    color: Optional[str] = None
    owner_id: UUID


class WorkspaceSummary(BaseModel):
    id: UUID
    name: str
    slug: str
    description: Optional[str] = None
    icon: Optional[str] = None
    color: Optional[str] = None

    class Config:
        from_attributes = True


class WorkspaceRead(WorkspaceSummary):
    organization_id: UUID
    settings: dict
    is_archived: bool
    created_by: UUID
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True

File: src/types/workspaces.ts
// Workspace types
export interface Workspace {
  id: string
  organization_id: string
  name: string
  slug: string
  description?: string
  color?: string
  icon?: string
  logo_url?: string
  settings: Record<string, any>
  is_archived: boolean
  created_by: string
  created_at: string
  updated_at: string
}

export interface WorkspaceSummary {
  id: string
  name: string
  slug: string
  color?: string
  icon?: string
}

export interface WorkspaceCreate {
  organization_id: string
  name: string
  slug: string
  description?: string
  color?: string
  icon?: string
  settings?: Record<string, any>
}

export interface WorkspaceUpdate {
  name?: string
  description?: string
  color?: string
  icon?: string
  logo_url?: string
  settings?: Record<string, any>
  is_archived?: boolean
}
